src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+# encoding: utf-8
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+#
+# Module: Manipulate system authentication configuration.
+# Author: Howard Guo <hguo@suse.com>
+
+require 'yast'
+require 'resolv'
+require 'open3'
+require 'socket'
+require 'tempfile'
+require 'fileutils'
+require 'date'
+
+module Auth
+    class AuthConf
```

Comment:
This is a new file of 1150 lines, which feels too many to me. It would be great if we found a way to split it up according to responsibilities.